FWCore/Modules/test/testGenericConsumer.py

```diff
@@ -0,0 +1,58 @@
+import FWCore.ParameterSet.Config as cms
+
+process = cms.Process("TEST")
+
+process.load("FWCore.Framework.test.cmsExceptionsFatal_cff")
+process.load("FWCore.MessageService.MessageLogger_cfi")
+process.MessageLogger.cerr.INFO.limit = 10000000
+
+process.maxEvents = cms.untracked.PSet(
+    input = cms.untracked.int32(5)
+)
+
+process.source = cms.Source("EmptySource")
+
+process.thing = cms.EDProducer("ThingProducer")
+
+process.anotherThing = cms.EDProducer("ThingProducer")
+
+process.otherThing = cms.EDProducer("OtherThingProducer",
+    thingTag = cms.InputTag('anotherThing'),
+    transient = cms.untracked.bool(True)
+)
+
+# a ConditionalTask makes the modules that are consumed on a path to be scheudled (only) in that path
+process.task = cms.ConditionalTask(
+    process.thing,
+    process.anotherThing,
+    process.otherThing
+)
+
+# test that the GenericConsumer will depend on 'thing' and 'otherThing', causing them to run
+process.consumer = cms.EDAnalyzer("GenericConsumer",
+    eventProducts = cms.untracked.vstring("*_thing_*_*", "otherThing"),
+    verbose = cms.untracked.bool(True)
+)
+
+# stop the processing of the path, preventing the analysers to run and trigger the unscheduled execution of 'thing' and 'anotherThing'
+process.stop = cms.EDFilter( "HLTBool",
+    result = cms.bool(False)
+)
```

Comment:
Could you replace this with e.g.
```suggestion
process.stop = cms.EDFilter( "TestFilterModule",
    acceptValue = cms.untracked.int32(-1)
)
```
? (we want to avoid adding a runtime dependence on non-framework packages)